src/datetime_rotation.jl

```diff
@@ -28,19 +33,25 @@ logger = DatetimeRotatingFileLogger(log_dir, raw"\a\c\c\e\s\s-yyyy-mm-dd.\l\o\g"
 logger = DatetimeRotatingFileLogger(log_dir, raw"yyyy-mm-dd-HH.\l\o\g") do io, args
     println(io, args.level, " | ", args.message)
 end
+
+# Example callback function to compress the recently-closed file
+callback(file) = run(`gzip $(file)`)
 """
 mutable struct DatetimeRotatingFileLogger <: AbstractLogger
     logger::Union{SimpleLogger,FormatLogger}
     dir::String
     filename_pattern::DateFormat
     next_reopen_check::DateTime
     always_flush::Bool
+    reopen_lock::ReentrantLock
+    current_file::Union{String,Nothing}
+    callback::Union{Function,Nothing}
```

Comment:
Is there a better name for this?

`on_rotation_callback` seems long.
Would `on_rotation` make sense?

Maybe `callback is best`

Alterantively, If we made it the first argument, we wouldn't have to have a name as part of the API

Reply:
`rotation_callback` maybe?

> Alterantively, If we made it the first argument, we wouldn't have to have a name as part of the API

What do you mean with this? For `do` purposes? This logger already has an optional format function for that so it is a bit awkward to have two.

Reply:
Ah true I forgot it already had one.
I wanted to avoid having to pick a name

`rotation_callback` works.
Makes it clear when it occurs (e.g. not a callback for every message)